web/access_control.go

```diff
@@ -150,11 +139,7 @@ func (s *AutograderService) getUserAndSCM(ctx context.Context, provider string)
 // All errors are logged, but only a single error is returned to the client.
 // This is a helper method to facilitate consistent treatment of errors and logging.
 func (s *AutograderService) getUserAndSCMForCourse(ctx context.Context, courseID uint64) (*pb.User, scm.SCM, error) {
```

Comment:
Turns out this is now just a forwarding to `getUserAndSCM`, which is used in:
- UpdateEnrollments
- UpdateGroup
- DeleteGroup
- IsEmptyRepo

To check if user is teacher, which should also be done in access control before calling these methods.